orchagent/dash/dashaclorch.cpp

```diff
@@ -0,0 +1,873 @@
+
+    if (!is_existing)
+    {
+        // Mandatory on create attributes should be setted, otherwise assign the default value.
+        // If the attributes don't have default value, just skip and wait for the user to set the value at the next message
+        if (!acl_rule.m_protocols)
+        {
+            const static vector<uint8_t> all_protocols = {
+                    0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,
+                    32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,
+                    64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,
+                    96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,
+                    128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,
+                    160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,
+                    192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,
+                    224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255
+                };
+            acl_rule.m_protocols = all_protocols;
+            attrs.emplace_back();
+            attrs.back().id = SAI_DASH_ACL_RULE_ATTR_PROTOCOL;
+            attrs.back().value.u8list.count = static_cast<uint32_t>(acl_rule.m_protocols.value().size());
+            attrs.back().value.u8list.list = acl_rule.m_protocols.value().data();
+        }
+
+        if (!acl_rule.m_priority || !acl_rule.m_dst_ips || !acl_rule.m_src_ips || !acl_rule.m_dst_ports || !acl_rule.m_src_ports)
+        {
+            SWSS_LOG_WARN("ACL rule %s doesn't have all mandatory attributes, waiting for user to set the value", key.c_str());
+            return task_success;
+        }
+    }
+
+    if (!attrs.empty())
+    {
+        if (!is_existing)
+        {
+            // Create a new ACL rule
+
+            attrs.emplace_back();
+            attrs.back().id = SAI_DASH_ACL_RULE_ATTR_DASH_ACL_GROUP_ID;
+            attrs.back().value.oid = acl_group->m_dash_acl_group_id;
+
+            acl_rule.m_status = m_dash_acl_rule_bulker.create_entry(&acl_rule.m_dash_acl_rule_id, static_cast<uint32_t>(attrs.size()), attrs.data());
+            acl_group->m_rule_count++;
+
+            m_defer_funcs.push_back([this, group_id, key]() {
+                if (m_dash_acl_rule_table[key].m_dash_acl_rule_id != SAI_NULL_OBJECT_ID)
+                {
+                    SWSS_LOG_NOTICE("Created ACL rule %s", key.c_str());
+                }
+                else
+                {
+                    getAclGroup(group_id)->m_rule_count--;
```

Comment:
Might be a good idea to add an error log here?

Reply:
Done